tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py

```diff
@@ -648,13 +651,17 @@ def _predict_tabular_data(self, new_data, process=True):
         if process:
             new_data = self._process_test_data(new_data)
         if not isinstance(new_data, TabularTorchDataset):
-            raise ValueError("new_data must of of type TabularTorchDataset if process=False")
+            raise ValueError("new_data must be of type TabularTorchDataset if process=False")
         val_dataloader = new_data.build_loader(self.max_batch_size, self.num_dataloading_workers, is_test=True)
         preds_dataset = []
         for data_batch in val_dataloader:
             preds_batch = self.model.predict(data_batch)
             preds_dataset.append(preds_batch)
-        preds_dataset = np.concatenate(preds_dataset, 0)
+        if len(preds_dataset) > 0:
+            preds_dataset = np.concatenate(preds_dataset, 0)
+        else:
+            num_net_outputs = self._get_num_net_outputs()
+            preds_dataset = np.zeros((0, num_net_outputs), dtype=np.float32)
```

Comment:
Is it common practice to support predicting on an empty dataset?
I think in general AutoGluon models crash in this scenario, so I'm unsure if we want to be supporting this case.

Happy to hear your thoughts.